polynote-server/src/main/scala/polynote/server/repository/NotebookRepository.scala

```diff
@@ -1,74 +1,224 @@
 package polynote.server.repository
 
-import java.io.FileNotFoundException
+import java.io.{File, FileNotFoundException}
 import java.nio.charset.StandardCharsets
 import java.nio.file.{FileAlreadyExistsException, FileVisitOption, Files, Path}
 
-import cats.effect.{ConcurrentEffect, ContextShift, IO, Sync}
-import cats.syntax.flatMap._
-import cats.syntax.functor._
-import cats.syntax.apply._
+import cats.implicits._
 import io.circe.Printer
-import org.http4s.client.blaze._
-import polynote.config.PolynoteConfig
+import polynote.config.{Mount, PolynoteConfig}
+import polynote.kernel.environment.{Config, Env}
+import polynote.kernel.{BaseEnv, GlobalEnv}
+import polynote.kernel.util.RefMap
 import polynote.messages._
 import polynote.server.repository.ipynb.ZeppelinNotebook
+import zio.{RIO, Task, ZIO}
+import zio.blocking.Blocking
+import zio.interop.catz._
 
 import scala.collection.JavaConverters._
 import scala.concurrent.ExecutionContext
 
-trait NotebookRepository[F[_]] {
+trait NotebookRepository {
 
   /**
     * @return Whether a notebook exists at the specified path
     */
-  def notebookExists(path: String): F[Boolean]
+  def notebookExists(path: String): RIO[BaseEnv with GlobalEnv, Boolean]
+
+  /**
+    * @return The location of the notebook on disk (absolute path)
+    */
+  def notebookLoc(path: String): RIO[BaseEnv with GlobalEnv, String]
```

Comment:
What if the notebook isn't on disk?

Reply:
That's a good point - I'll update it to be a URI.

Reply:
Updated!